src/wp-includes/theme.php

```diff
@@ -4220,6 +4220,31 @@ function create_initial_theme_features() {
 			),
 		)
 	);
+	register_theme_feature(
+		'editor-shadow-presets',
+		array(
+			'type'         => 'array',
+			'description'  => __( 'Custom shadow presets if defined by the theme.' ),
+			'show_in_rest' => array(
+				'schema' => array(
+					'items' => array(
+						'type'       => 'object',
+						'properties' => array(
+							'name'   => array(
+								'type' => 'string',
+							),
+							'shadow' => array(
+								'type' => 'string',
+							),
+							'slug'   => array(
+								'type' => 'string',
+							),
+						),
+					),
+				),
+			),
+		)
+	);
```

Comment:
Bonus support for custom shadows in classic themes to match other presets.

Reply:
This was required in order to support these lines and match the behavior of colors/gradients.

https://github.com/WordPress/wordpress-develop/pull/6295/files#diff-d6b86476eed058e7cf8b6e57fa52c4fd75b1f0907e1a9ccb0149528a24f7578bR322-R325

```php
if ( ! isset( $theme_support_data['settings']['shadow']['presets'] ) ) {
	// If the theme does not have any shadows, we still want to show the core ones.
	$default_shadows = true;
}
```

If we decide to diverge from the colors/gradients behavior like we're discussing in https://github.com/WordPress/wordpress-develop/pull/6295/files#r1533015533, we could probably remove it.

Reply:
I opened an alternative PR (https://github.com/WordPress/wordpress-develop/pull/6303) with shadows enabled by default in classic themes (like colors/gradients) but without the option to opt-out with `add_theme_support( 'editor-shadow-presets', array() )`.

Both PRs fix the regression that I'm concerned with.